c10/core/SymNodeImpl.h

```diff
@@ -205,6 +217,7 @@ class C10_API SymNodeImpl : public c10::intrusive_ptr_target {
     os << str();
     return os;
   }
+  c10::DispatchKeySet key_set_;
```

Comment:
This is an interesting choice. It is the optimal choice from a performance perspective as it means we can fetch the key set for all SymNodes directly without having to perform a vcall, but it also means the correct key set must be known at SymNode construction time (as you have imposed a constraint on the layout of all SymNode objects). On balance, this is probably the right way to do it, but could we at least make the field protected and put an (inlineable) accessor to make it easier to change implementation decisions in the future?